www/advanced-data-model.html

```diff
@@ -0,0 +1,621 @@
+                // collision with API
+                let context = imageJob.userData;
+                context.image = new Image();
+                
+                // in all scenarios, unless abort() is called, make
+                // sure imageJob.finish() gets executed!
+                context.image.onerror = context.image.onabort = function() {
+                    imageJob.finish(null, context.promise, "Failed to parse tile data as an Image");
+                };
+                context.image.onload = function() {
+                    imageJob.finish(context.image, context.promise);
+                };
+                
+                function createErrorImageURL(e) {
+                    let canvas = document.createElement('canvas');
+                    let context = canvas.getContext('2d');
+                    
+                    //yay, postData = tile instance
+                    let tile = imageJob.postData;
+                    canvas.width = tile.width;
+                    canvas.height = tile.height;
+                    context.font = "80px Georgia";
+                    context.fillStyle = "#ff2200";
+                    context.fillText(e, 5, 120);
+                    return canvas.toDataURL("image/jpeg");
+                }
+                            
+                // note we ignore some imageJob properties such as 
+                // 'loadWithAjax'. This means preventing OSD from using
+                // ajax will have no effect as we force it to do so.
+                // Make sure you implement all the features the official
+                // implementation do if you want to keep them.
+                context.promise = myFetch(imageJob.src, {
+                    method: "GET",
+                    mode: 'cors',
+                    cache: 'no-cache',
+                    credentials: 'same-origin',
+                    headers: imageJob.ajaxHeaders || {},
+                    body: null
+                }).then(data => {
+                    //to spice up things, emulate faulty source
+                    if (Math.random() > 0.7) throw "Oh no!";
+                    return data.blob();
+                }).then(blob => {
+                    context.image.src = URL.createObjectURL(blob);
+                }).catch(e => {
+                    context.image.src = createErrorImageURL(e);
+                }); 
+            
+            },
+            downloadTileAbort: function(imageJob) {
+                //we can abort thanks to myFetch() implementation
+                imageJob.userData.promise.abort();
+            } 
+        });
+         
+    OpenSeadragon({
+        id:            "example-custom-tilesource-advanced",
+        prefixUrl:     "/openseadragon/images/",
+        navigatorSizeRatio: 0.25,
+        wrapHorizontal:     true,
+        loadTilesWithAjax:  true, // no effect
+        tileSources: [
+            "/example-images/highsmith/highsmith.dzi"
```

Comment:
If you use https://openseadragon.github.io/example-images/highsmith/highsmith.dzi, it'll work even when run locally (and will be easier for people to try on their own machines).

Reply:
The problem is a localhost domain differs from the upstream domain. It results in the CORS policy violation.

Reply:
Oh, I see! Fair enough. You tested it, I assume? I think some CORS policies allow different domains, but I guess github.io does not. In that case, better to leave it as it is.

Reply:
Yes, I tested it. That's why the document contained `... to run into problems such as CORS policy. Guess what happened when writing the tutorial.`